frontend/src/pages/Compare.tsx

```diff
@@ -47,23 +58,68 @@ export default function Compare(props: CompareProps) {
           mode: 'error',
         });
       },
-      onSuccess: () => props.updateBanner({}),
+      onSuccess: data => {
+        // Set the version based on the runs included.
+        let version: CompareVersion = CompareVersion.InvalidRunCount;
+        if (data && data.length >= 2 && data.length <= 10) {
+          for (const run of data) {
+            const runVersion = run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest')
+              ? CompareVersion.V2
+              : CompareVersion.V1;
+            if (version === CompareVersion.InvalidRunCount) {
+              version = runVersion;
+            } else if (version !== runVersion) {
+              version = CompareVersion.Mixed;
+            }
+          }
+        }
```

Comment:
Optional: How about the following logic? Feel free to make a decision whether it helps with readability.

```
if (data && data.length >= 2 && data.length <= 10) {
    setCompareVersion(CompareVersion.InvalidRunCount)
}
const v2runs = data.filter(run => run.run?.pipeline_spec?.hasOwnProperty('pipeline_manifest'));
if (v2runs.length == 0) {
    setCompareVersion(CompareVersion.V1)
} else if (v2runs.length == data.length) {
    setCompareVersion(CompareVersion.V2)
} else {
    setCompareVersion(CompareVersion.Mixed)
}


```

Reply:
Yes, I think this is better - I updated that in this commit: [`f2dea93`](https://github.com/kubeflow/pipelines/pull/7844/commits/f2dea93d21703bf1d8290425ecaa5e993b682542#diff-0a82aa9abe0f60d0f25f94d6886f8e8f86a1ccc698baef2c540b9b0941d5f4c0).

The only change was the above code is the top line - I removed the check to see if `data` is defined as it must be if used in the [`onSuccess`](https://react-query.tanstack.com/reference/useQuery) callback, and since I'm checking to see if the run count is _invalid_, the line became the following:
```
if (data.length < 2 || data.length > 10) {
```

Reply:
Yes you are right! Thanks for catching the issue in data length check condition!